linkerd/stack/src/failfast.rs

```diff
@@ -163,24 +340,45 @@ where
     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         match self.project() {
             ResponseFutureProj::Inner(f) => f.try_poll(cx).map_err(Into::into),
-            ResponseFutureProj::FailFast(scope) => Poll::Ready(Err(FailFastError { scope }.into())),
+            ResponseFutureProj::FailFast => Poll::Ready(Err(FailFastError(()).into())),
         }
     }
 }
 
+// === impl Shared ===
+
+impl Shared {
+    fn exit_failfast(&self) {
+        // The load part of this operation can be `Relaxed` because this task
+        // is the only place where the the value is ever set.
+        if self
+            .in_failfast
+            .compare_exchange(true, false, Ordering::Release, Ordering::Relaxed)
+            .is_ok()
+        {
```

Comment:
This seems simpler and equivalent, right?

```suggestion
        if self.in_failfast.swap(false, Ordering::Release) {
            self.notify.notify_waiters();
        }
```

Reply:
ah, yeah, it is, good suggestion!